File: Java/src/main/java/p1004.java
import java.util.ArrayList;
import java.util.Scanner;

public class p1004 {

	public static void main(String[] args) {
		Scanner scanner = new Scanner(System.in);

		ArrayList<Integer> result = new ArrayList<Integer>();

		int x1, y1;
		int x2, y2;
		int x3, y3, r3;
		int planetCount;
		int resultCount = 0;
		double distanceStart, distanceArrive;
		int testCase = scanner.nextInt();

		while(testCase != 0) {
			x1 = scanner.nextInt();
			y1 = scanner.nextInt();
			x2 = scanner.nextInt();
			y2 = scanner.nextInt();

			planetCount = scanner.nextInt();

			for(int i = 0; i < planetCount; i++) {
				x3 = scanner.nextInt();
				y3 = scanner.nextInt();
				r3 = scanner.nextInt();

				distanceStart = Math.sqrt(Math.pow(x1 -x3, 2) + Math.pow(y1-y3, 2));
				distanceArrive = Math.sqrt(Math.pow(x2 -x3, 2) + Math.pow(y2-y3, 2));

				if((r3 > distanceArrive && r3 < distanceStart)
						|| (r3 < distanceArrive && r3 > distanceStart)) {
					resultCount++;
				}
			}

			result.add(resultCount);
			resultCount = 0;
			testCase--;
		}

		for(int i = 0; i < result.size(); i++) System.out.println(result.get(i));
	}
}

File: C++/C++/p1004.cpp
//
//  p1004.cpp
//  C++
//
//  Created by KimHanJu on 09/09/2019.
//  Copyright © 2019 KimHanJu. All rights reserved.
//

#include <iostream>
#include <vector>
#include <cmath>
using namespace std;

double getDistance(int x, int y, int planetX, int planetY) {
    return sqrt(pow(x - planetX, 2) + pow(y - planetY, 2));
}

int main(int argc, const char * argv[]) {
    vector<int> result;
    
    int testCase;
    int x1, y1;
    int x2, y2;
    int x3, y3, r3;
    int planetCount, resultCount = 0;
    double distanceStart, distanceArrive;
    
    cin >> testCase;
    
    while(testCase --> 0) {
        cin >> x1 >> y1 >> x2 >> y2;
        cin >> planetCount;
        
        while(planetCount --> 0) {
            cin >> x3 >> y3 >> r3;
            distanceStart = getDistance(x1, y1, x3, y3);
            distanceArrive = getDistance(x2, y2, x3, y3);
            
            if((r3 > distanceArrive && r3 < distanceStart)
               || (r3 < distanceArrive && r3 > distanceStart)) {
                resultCount++;
            }
        }
        
        result.push_back(resultCount);
        resultCount = 0;
    }

    for(int value : result) cout << value << "\n";
    
    return 0;
}
